pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentTopic.java

```diff
@@ -683,12 +683,8 @@ public void updateSubscribeRateLimiter() {
     }
 
     private void asyncAddEntry(ByteBuf headersAndPayload, PublishContext publishContext) {
-        if (brokerService.isBrokerEntryMetadataEnabled()) {
-            ledger.asyncAddEntry(headersAndPayload,
-                    (int) publishContext.getNumberOfMessages(), this, publishContext);
-        } else {
-            ledger.asyncAddEntry(headersAndPayload, this, publishContext);
-        }
+        ledger.asyncAddEntry(headersAndPayload,
```

Comment:
You also need to handle publish Txn messages
https://github.com/apache/pulsar/blob/2d46bfafab2567593067847775b2d672ed502144/pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentTopic.java#L4236

Reply:
Since the transaction part needs to refactor the interface. So I didn't include it in this cleanup. And the numberOfMessages is only used for the interceptor. So it should be fine. We can use another PR to resolve the transaction issue.